### PR TITLE
add Order entity and impl its functionality, made changes in Shopping…

### PR DESCRIPTION
**Pull request by NataliaKurhanska**
…Cart functionality

### CODE REVIEW

**Comment by NataliaKurhanska** [comment]
Hi! I also made some changes in ShoppingCart functionality, please, check

File: src/main/java/bookstore/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+@RequestMapping("/api/orders")
+@RequiredArgsConstructor
+public class OrderController {
+    private final OrderService orderService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @Operation(summary = "Place an order",
+            description = "Place an order and clear the shopping cart")
+    public OrderDto createOrder(Authentication authentication, @RequestBody String shippingAdress) {
```
Comment (sokimaaa) [inline src/main/java/bookstore/controller/OrderController.java]: Create object for storing shipping address 
```suggestion
    public OrderDto createOrder(Authentication authentication, Object///) {
```

Comment (sokimaaa) [inline src/main/java/bookstore/controller/OrderController.java]: Consider using ResponseEntity<?> in controller 
```suggestion
    public ResponseEntity<OrderDto> createOrder(Authentication authentication, @RequestBody String shippingAdress) {
```

File: src/main/java/bookstore/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    @Operation(summary = "Get order item by its id", description = "Get order item by its id")
+    public OrderItemDto getOrderItemById(@PathVariable Long orderId, @PathVariable Long itemId) {
+        return orderService.getOrderItemById(orderId, itemId);
+    }
+
+    @PatchMapping("/{id}")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(summary = "Update order's status", description = "Update order's status by admin")
+    public OrderDto updateOrderStatus(@PathVariable Long id,
+                                      @RequestParam Order.Status status) {
```
Comment (sokimaaa) [inline src/main/java/bookstore/controller/OrderController.java]: check DTO pattern, why do we need it, and place here dto respectively
```suggestion
                                      @RequestParam Dto status) {
```

File: src/main/java/bookstore/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -55,16 +58,17 @@ public void addItemtoShoppingCart(Authentication authentication,
...
         return shoppingCartService.updateCartItemsQuantity(cartItemId, quantity);
     }
 
     @DeleteMapping("/cart-items/{cartItemId}")
     @Operation(summary = "Delete book from shopping cart",
             description = "Delete book from user's shopping cart")
     @PreAuthorize("hasRole('ROLE_USER')")
-    public void deleteItemFromCart(Authentication authentication,
-                                   @PathVariable Long cartItemId) {
+    public void deleteItemFromCart(Authentication authentication, @PathVariable Long cartItemId) {
```
Comment (sokimaaa) [inline src/main/java/bookstore/controller/ShoppingCartController.java]: never return void from controller

File: src/main/java/bookstore/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderItemDto> getOrderItemsByOrder(Long orderId) {
+        Order order = orderRepository.findById(orderId).orElseThrow(() ->
+                new EntityNotFoundException("Can't find order by id: " + orderId));
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .collect(Collectors.toList());
```
Comment (sokimaaa) [inline src/main/java/bookstore/service/impl/OrderServiceImpl.java]: inline expression
```suggestion
        Order order = orderRepository.findById(orderId).orElseThrow(() ->
                new EntityNotFoundException("Can't find order by id: " + orderId));
        return order.getOrderItems().stream()
                .map(orderItemMapper::toDto)
                .collect(Collectors.toList());
```

File: src/main/java/bookstore/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    @Override
+    public OrderItemDto getOrderItemById(Long orderId, Long itemId) {
+        return getOrderItemsByOrder(orderId).stream()
```
Comment (sokimaaa) [inline src/main/java/bookstore/service/impl/OrderServiceImpl.java]: bad solution, why do we need extract from database all orders and filter it here, if we can directly extract from database needed order

File: src/main/java/bookstore/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                .filter(i -> i.getId().equals(itemId))
+                .findFirst()
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order item by id: "
+                        + itemId));
+    }
+
+    private Set<OrderItem> getOrderItems(ShoppingCart shoppingCart, Order order) {
+        Set<OrderItem> orderItems = shoppingCart.getCartItems().stream()
+                .map(cartItemOrderMapper::toOrderItem)
+                .collect(Collectors.toSet());
```
Comment (sokimaaa) [inline src/main/java/bookstore/service/impl/OrderServiceImpl.java:88]: consider Stream<T>.peek, Stream<T>.forEach

**Pull request merged**